### PR TITLE
Add rally-ovs run create_network tasks to docker ovn emulation

### PR DESCRIPTION
**Pull request by huikang**
- Add install_method in network_create_args
- generate rally task files from template in rally role
- update README about create_network rally job

Signed-off-by: Hui Kang kangh@us.ibm.com

### CODE REVIEW

**Comment by huikang** [comment]
Now we can run create_network workload to the docker-based emulation environment.

File: rally_ovs/plugins/ovs/scenarios/ovn.py [hunk 1]
```diff
@@ -44,9 +44,14 @@ def _create_lswitches(self, lswitch_create_args):
         start_cidr = lswitch_create_args.get("start_cidr", "")
         if start_cidr:
             start_cidr = netaddr.IPNetwork(start_cidr)
+        install_method = lswitch_create_args.get("install_method", None)
+        if not install_method:
+            print "No install_method, use default"
+            install_method = "sandbox"
+        print "create network method: ", install_method
```
Comment (l8huang) [inline rally_ovs/plugins/ovs/scenarios/ovn.py]: replace 'print' with a info log

File: rally_ovs/plugins/ovs/scenarios/ovn.py [hunk 2]
```diff
@@ -44,9 +44,14 @@ def _create_lswitches(self, lswitch_create_args):
         start_cidr = lswitch_create_args.get("start_cidr", "")
         if start_cidr:
             start_cidr = netaddr.IPNetwork(start_cidr)
+        install_method = lswitch_create_args.get("install_method", None)
```
Comment (l8huang) [inline rally_ovs/plugins/ovs/scenarios/ovn.py]: install_method = lswitch_create_args.get("install_method", "sandbox")

File: rally_ovs/plugins/ovs/scenarios/ovn.py [hunk 3]
```diff
@@ -230,9 +235,15 @@ def _create_networks(self, network_create_args):
         physnet = network_create_args.get("physical_network", None)
         lswitches = self._create_lswitches(network_create_args)
 
+        install_method = network_create_args.get("install_method", None)
+        if not install_method:
+            print "No install_method, use default"
+            install_method = "sandbox"
```
Comment (l8huang) [inline rally_ovs/plugins/ovs/scenarios/ovn.py]: install_method = lswitch_create_args.get("install_method", "sandbox")
replace 'print' with a info log

File: samples/tasks/scenarios/ovn-sandbox/create_sandbox_k12c0ga.json [hunk 4]
```diff
@@ -7,10 +7,10 @@
             "name": "OvnSandbox.create_sandbox",
             "args": {
                 "sandbox_create_args": {
-                    "farm": "ovn-farm-node-0",
```
Comment (l8huang) [inline samples/tasks/scenarios/ovn-sandbox/create_sandbox_k12c0ga.json:10]: json files under 'samples' folder are simply samples, we need not commit environment-specific json files into it.
Seem like 'create_sandbox_k12c0ga.json' is a task file dedicated to your environment, you may keep it in a private repo.

File: rally_ovs/plugins/ovs/scenarios/ovn.py (hunk 3 above)
Comment (huikang) [inline rally_ovs/plugins/ovs/scenarios/ovn.py]: if we use `install_method = lswitch_create_args.get("install_method", "sandbox")`, I think we can remove `install_method = "sandbox"`.
Correct?

File: samples/tasks/scenarios/ovn-sandbox/create_sandbox_k12c0ga.json (hunk 4 above)
Comment (huikang) [inline samples/tasks/scenarios/ovn-sandbox/create_sandbox_k12c0ga.json:10]: Will remove. Add these files by mistake. Thanks.

**Comment by l8huang** [comment]
samples/tasks/scenarios/ovn-sandbox/create_sandbox_k12c0ga.json
samples/tasks/scenarios/ovn-sandbox/delete_sandbox_k12c0ga.json

The two files don't show any new functionality, needn't push them into samples folder, you may keep them locally or in another repo.

File: rally_ovs/plugins/ovs/scenarios/ovn.py (hunk 3 above)
Comment (l8huang) [inline rally_ovs/plugins/ovs/scenarios/ovn.py]: Yes, get() will return "sandbox" if "lswitch_create_args" has no "install_method" in it.

**Pull request merged**